record-core/src/androidMain/kotlin/dev/theolm/record/RecordCore.android.kt

```diff
@@ -4,60 +4,103 @@ package dev.theolm.record
 
 import android.Manifest.permission.RECORD_AUDIO
 import android.content.pm.PackageManager.PERMISSION_GRANTED
+import android.media.AudioRecord
 import android.media.MediaRecorder
 import android.os.Build
 import androidx.core.content.ContextCompat
-import dev.theolm.record.config.AudioEncoder
 import dev.theolm.record.config.OutputFormat
-import dev.theolm.record.config.OutputLocation
 import dev.theolm.record.config.RecordConfig
 import dev.theolm.record.error.NoOutputFileException
 import dev.theolm.record.error.PermissionMissingException
 import dev.theolm.record.error.RecordFailException
+import java.io.FileOutputStream
 
 @Suppress("TooGenericExceptionCaught", "SwallowedException")
 internal actual object RecordCore {
+    private var audioRecord: AudioRecord? = null
+    private var recordingThread: Thread? = null
     private var recorder: MediaRecorder? = null
     private var output: String? = null
-    private var recordingState : RecordingState = RecordingState.IDLE
+    private var myRecordingState : RecordingState = RecordingState.IDLE
 
     @Throws(RecordFailException::class)
     internal actual fun startRecording(config: RecordConfig) {
         checkPermission()
         output = config.getOutput()
-        recorder = createMediaRecorder(config)
+        when(config.outputFormat) {
+            OutputFormat.MPEG_4 -> {
+                recorder = createMediaRecorder(config)
 
-        recorder?.apply {
-            runCatching {
-                prepare()
-            }.onFailure {
-                throw RecordFailException()
-            }
+                recorder?.apply {
+                    runCatching {
+                        prepare()
+                    }.onFailure {
+                        throw RecordFailException()
+                    }
+
+                    setOnErrorListener { mr, what, extra ->
```

Comment:
`mr, what, extra` hahaha

you are not using this params, please remove them or give them a proper name.

```kt
setOnErrorListener { _, _, _ ->

```